hello-server.ts

```diff
@@ -0,0 +1,13 @@
+import { createServer, IncomingMessage, ServerResponse } from "http";
+
+function handler(req: IncomingMessage, res: ServerResponse) {
+  res.writeHead(200, { "Content-Type": "text/plain" });
+  res.end(`Hello, world!`);
+}
+
+const server = createServer(handler);
+
+// Start the server
+server.listen(3000, function() {
```

Comment:
Passing a function to another function as the last argument is a common pattern and called a "callback." We do this in order to say "once you're done, do this." In this case, start listening for requests and when you're done setting up, let us print to the terminal that we're ready to go.